File: TypeScriptDataStructure/SinglyLinkedList/ListNode.ts
export class ListNode {
  val: number
  next: ListNode | null

  constructor(val?: number, next?: ListNode | null) {
    this.val = val
    this.next = next
  }

  static create(arr: number[]) {
    const dummy = new ListNode(0)
    let temp = dummy
    for (const num of arr) {
      temp.next = new ListNode(num)
      temp = temp.next
    }
    return dummy.next
  }
}

File: RubyDataStructure/SinglyLinkedList/ListNode.rb
class ListNode
  attr_accessor :next, :val
  def initialize(val=0, next_node=nil)
    @val = val
    @next = next_node
  end

  def self.create(arr)
    dummy = ListNode.new(0)
    temp = dummy
    arr.each do |num|
      temp.next = ListNode.new(num)
      temp = temp.next
    end
    return dummy.next
  end
end